src/compiler/scala/tools/nsc/transform/Delambdafy.scala

```diff
@@ -244,7 +254,7 @@ abstract class Delambdafy extends Transform with TypingTransformers with ast.Tre
         }
 
         val bridge = postErasure.newTransformer(unit).transform(DefDef(methSym, List(bridgeParams.map(ValDef(_))),
-          adaptToType(forwarderCall setType functionResultType, bridgeResultType))).asInstanceOf[DefDef]
+          adaptToType(forwarderCall.setType(forwarderType), bridgeResultType))).asInstanceOf[DefDef]
```

Comment:
IIUC the issue you're addressing is the same (in opposite direction) as the one addressed 10 lines above for arguments. I like that this existing fix is more local.

Could we do this instead?

```
        val forwarderActualType =
          if (samResultType.isInstanceOf[ErasedValueType] && functionResultType.isInstanceOf[ErasedValueType]) bridgeResultType
          else functionResultType
```

and then 
```suggestion
          adaptToType(forwarderCall.setType(forwarderActualType), bridgeResultType))).asInstanceOf[DefDef]
```

Reply:
Yes, I didn't simplify the expression.